tasks.php

```diff
@@ -1901,3 +1906,12 @@ function title_string_for_task($pre_task)
 {
     return sprintf(_("Task #%d: %s"), $pre_task->task_id, $pre_task->task_summary);
 }
+
+// Get key corresponding to a task properties string
+// Input string must be a valid property value
+function get_property_key($value, $array)
```

Comment:
It looks like the value this function has over `array_search()` is that it checks to see if the value exists and fails if it doesn't, but it does so with `assert()` which are often [disabled in production](https://www.php.net/manual/en/ini.core.php#ini.zend.assertions) rendering this effectively the same thing. If we really want to fail on an invalid error we should throw a `ValueError` exception instead of using `assert()`.

I was also wondering if this should be in `misc.inc` as `array_get_key()` to mirror `array_get()` but the value in the latter is the ability to include a default which I don't think we really want to do here so I'd opt for keeping it here with the name you have and just throwing on error.

Reply:
If there's a need in the future for a more general `array_get_key()`, it can be renamed and moved to `misc.inc`, I suppose.
I've made the change you suggested.